arches/app/media/js/utils/map-popup-provider.js

```diff
@@ -40,6 +41,45 @@ define(['arches',
             return features;
         },
 
+        /**
+         * This method enables custom logic for how the feature in the popup should be handled and/or mutated en route to the mapFilter.
+         * @param popupFeatureObject - the javascript object of the feature and its associated contexts (e.g. mapCard).
+         * @required @method mapCard.filterByFeatureGeom()
+         * @required @send argument: @param feature - a geojson feature object 
+         */
+        sendFeatureToMapFilter: function(popupFeatureObject)
+        {
+            let foundFeature = null;
+            const strippedFeatureId = popupFeatureObject.feature.properties.featureid.replace(/-/g,"");
+            for (let geometry of popupFeatureObject.geometries()) {
+                if (geometry.geom && Array.isArray(geometry.geom.features)) {
+                    foundFeature = geometry.geom.features.find(feature => feature.id.replace(/-/g, "") === strippedFeatureId);
+                    if (foundFeature)
+                        break;
+                }
+            }
+            popupFeatureObject.mapCard.filterByFeatureGeom(foundFeature);
+        },
+
+        /**
+         * Determines whether to show the button for Filter By Feature
+         * @param popupFeatureObject - the javascript object of the feature and its associated contexts (e.g. mapCard).
+         * @returns {boolean} - whether to show "Filter by Feature" on map popup
+         * typically dependent on at least 1 feature with a geometry and/or a featureid/resourceid combo
+         */
+        showFilterByFeature: function(popupFeatureObject) {
+            let foundFeature = null;
+            const strippedFeatureId = popupFeatureObject.feature.properties.featureid.replace(/-/g,"");
```

Comment:
This throws and exception if the feature doesn't have a featureid. Suggest doing a check for existence first and return false if there is no featureid.

Perhaps refactor to extract common logic that searches for the feature to avoid code duplication? (`findSelectedFeature()`?)